eclipse/src/saros/session/resources/validation/ResourceChangeValidator.java

```diff
@@ -60,22 +64,34 @@ public IStatus validateChange(final IResourceDelta delta, final IProgressMonitor
 
     if (currentSession == null) return Status.OK_STATUS;
 
-    final ResourceDeltaVisitor visitor = new ResourceDeltaVisitor(currentSession);
+    Set<IReferencePoint> sharedReferencePoints = session.getReferencePoints();
 
-    try {
-      delta.accept(visitor);
-    } catch (CoreException e) {
-      log.warn(
-          "error occured during delta visiting, some resources might have not been checked", //$NON-NLS-1$
-          e);
-    }
+    for (IReferencePoint referencePoint : sharedReferencePoints) {
+      IContainer referencePointDelegate = ResourceConverter.getDelegate(referencePoint);
+
+      IResourceDelta referencePointDelta = delta.findMember(referencePointDelegate.getFullPath());
+
+      final ResourceDeltaVisitor visitor = new ResourceDeltaVisitor(referencePoint, currentSession);
 
-    if (!currentSession.hasWriteAccess() && visitor.isModifyingResources())
-      return MODIFYING_RESOURCES_ERROR_STATUS;
+      if (referencePointDelta != null) {
```

Comment:
From the documentation:

The supplied path may be absolute or relative; in either case, it is interpreted as relative to this delta. Trailing separators are ignored. If the path is empty this delta is returned. 

So is /Foo/Bar/Foo (fullpath) applied on the Delta (/Foo/Bar) checked as /Foo/Bar/Foo or /Foo/Bar/Foo/Bar/Foo ?

Reply:
The way I understand it, the initial delta always start at the workspace root. So, as we are always working on the base delta here, using the full path should result in the correct directory.

But I will check if this assumption holds true. If it does not, I can adjust the logic to get the resource of the delta, get its full path and then compute the relative path between the base resource and the reference point resource (if there is one).

Reply:
From my testing, the initial delta always starts at the workspace root. Any non-changed resources are marked as such in the delta. But I could not find any official documentation on this behavior.

So if you prefer, I can still add the logic to create the relative path between the delta resource and the reference point delegate and use it instead anyways just to make sure.